src/Command/Check.hs

```diff
@@ -50,11 +51,41 @@ optionsParser = Options
     <> help "Write logs to FILE instead of standard output"
     <> action "file"
     )
+  <*> switch (
+       long "watch"
+    <> help "Watch the files for any changes, re-typechecking as needed."
+    )
+
+check :: Options -> IO ()
+check opts = if watch opts then checkWatch opts else join (checkSimple opts)
+
+checkWatch :: Options -> IO ()
+checkWatch opts = do
+  (dirs, files) <- splitDirsAndFiles (inputFiles opts)
+  _exit <- checkSimple opts
+  go dirs myWatchDir $
+    go files myWatchFile $
+      forever (threadDelay oneSecond)
+  where
+    go :: [a] -> (WatchManager -> a -> IO b) -> IO c -> IO c
+    go (p:ps) f end = withManager (\wm -> f wm p >> go ps f end)
```

Comment:
Could we reuse the same manager for all files and directories we want to watch, with a single `withManager` call around the two calls to `go`, or are they restricted to one directory?

Reply:
It seems to be undocumented but I tried multiple things and it works :)